zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/Publisher.java

```diff
@@ -117,61 +129,143 @@ public static boolean Publish(Collection collection, String email, CollectionRea
      *
      * @param collection The collection to be published.
      * @param email      An identifier for the publishing user.
      * @return If publishing succeeded, true.
      * @throws IOException If a general error occurs.
      */
-    public static boolean PublishFilesToWebsite(Collection collection, String email, CollectionReader collectionReader) throws IOException {
+    private boolean FullPublish(Collection collection, String email, CollectionReader collectionReader) {
 
         boolean publishComplete = false;
         String encryptionPassword = Random.password(100);
         try {
-            collection.description.publishStartDate = new Date();
-            BeginPublish(collection, encryptionPassword);
-            SendManifest(collection, encryptionPassword);
-            PublishFilteredCollectionFiles(collection, collectionReader, encryptionPassword);
-            publishComplete = CommitPublish(collection, email, encryptionPassword);
-            collection.description.publishEndDate = new Date();
+
+            DoPrePublish(collection, encryptionPassword);
+            publishComplete = DoPublish(collection, collectionReader, encryptionPassword, email);
 
         } catch (IOException e) {
+            logError(e, "Exception publishing collection").collectionName(collection).log();
+        }
+
+        return publishComplete;
+    }
+
+    private static boolean publishFiles(Collection collection, CollectionReader collectionReader, String encryptionPassword, String userEmail) throws IOException {
+
+        boolean filesPublished = false;
+
+        try {
+            collection.description.publishStartDate = new Date();
+            SendFilteredCollectionFiles(collection, collectionReader, encryptionPassword);
+            filesPublished = CommitPublish(collection, userEmail, encryptionPassword);
 
+        } catch (IOException e) {
             SlackNotification.alarm(String.format("Exception publishing collection: %s: %s", collection.description.name, e.getMessage()));
-            logError(e, "Exception publishing collection").collectionName(collection).log();
+            logError(e, "exception publishing collection").collectionName(collection).log();
+
             // If an error was caught, attempt to roll back the transaction:
-            Map<String, String> transactionIds = collection.description.publishTransactionIds;
-            if (transactionIds != null && transactionIds.size() > 0) {
-                logInfo("Attempting rollback of collection publishing transaction").collectionName(collection).log();
-                rollbackPublish(collection.description.publishTransactionIds, encryptionPassword);
+            Map<String, String> publishTransactionIds = collection.description.publishTransactionIds;
+            if (publishTransactionIds != null && publishTransactionIds.size() > 0) {
+                logInfo("attempting rollback of collection publishing transaction").collectionName(collection).log();
```

Comment:
is it worth adding the transactionID to this log message?

Reply:
Done